common/common.go

```diff
@@ -0,0 +1,10 @@
+package common
+
+func ApplyOption[T any](instance *T, options []func(*T) error) (*T, error) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Very elegant !